cni/pkg/nodeagent/informers.go

```diff
@@ -212,19 +216,42 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 		wasAnnotated := oldPod.Annotations != nil && oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		shouldBeEnabled := util.PodRedirectionEnabled(ns, newPod)
+		wasTerminated := kube.CheckPodTerminal(oldPod)
+		isTerminated := kube.CheckPodTerminal(newPod)
+		// only checks if the pod state just changed from not-terminated to terminated.
+		// if pod state changed in *any* other way, this should be false.
+		justTerminated := !wasTerminated && isTerminated
```

Comment:
Do we need this? In general I think its a good practice for controllers to act on the current state. For example, what if we restart -- we do not get the old state.

I think we handle this case since startup are different, so this is more a style comment than a bug I think. For instance, though, we could remove the annotation. Then the check is just "is annotated && is terminated => remove"

Reply:
We could but then we'd need to cover the inverse state the same way if we needed to re-annotate because the pod status changed back.

(Or at least that was the intent - making these conditionals overbroad tends to create state handling corner cases)

Reply:
Ah right - won't change back.